pkg/apis/clusterregistry/v1alpha1/types.go

```diff
@@ -167,3 +171,87 @@ type ClusterCondition struct {
 	// +optional
 	Message string `json:"message,omitempty" protobuf:"bytes,6,opt,name=message"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCredentials combines a Cluster record with a set of credentials
+// (stored in a kubeconfig) that can be used to access the cluster. The status
+// of ClusterCredentials reflects whether those credentials can be used - from
+// within the cluster hosting the ClusterCredentials resource - to access the
+// cluster's healthz endpoint.
+//
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=clustercredentials
+// +kubebuilder:subresource:status
+type ClusterCredentials struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterCredentialsSpec   `json:"spec,omitempty"`
+	Status ClusterCredentialsStatus `json:"status,omitempty"`
+}
+
+// ClusterCredentialsSpec defines the desired state of a ClusterCredentials.
+type ClusterCredentialsSpec struct {
+	// ClusterRef is a reference to the Cluster to access.
+	ClusterRef v1.LocalObjectReference `json:"clusterRef,omitempty"`
```

Comment:
If this were an object reference, it would be easier to have a cluster registry with a public list of clusters (in, say, `kube-multicluster-public`) and have each user/team have their own ClusterConnection in their own namespace. That would make access control easier to reason about.

Reply:
That's a very good point, thanks for the feedback. I will add a distinct option for a namespace-scoped reference